File: solutions/lua/pangram/pangram.lua
return function(sentence)
    local letters = {}
    local found = 0

    for char in sentence:lower():gmatch("%l") do
        if not letters[char] then
            letters[char] = true
            found = found + 1
        end
        if found == 26 then
            return true
        end
    end

    return false
end

File: solutions/r/pangram/pangram.R
# Determines if a word or sentence is a pangram.
# A pangram is a sentence using every letter of the alphabet at least once.
is_pangram <- function(input) {
    (charlist <- input
        |> tolower()
        |> strsplit("")
        |> unlist())

    all(letters %in% charlist)
}
